src/server/ServerManager.cpp

```diff
@@ -88,18 +88,25 @@ void ServerManager::mapServerToSocket(
   }
 }
 
-const Server* ServerManager::getServerFromSocket(const Socket* socket) const
+/*
+  Tries to find the best server for the given hostname. If no hostname matches
+  it defaults to the first server that is associated with this socket.
+*/
+const Server* ServerManager::getServerByHost(const Socket* socket,
+                                             const std::string& host) const
 {
-  if (socket == FT_NULLPTR) {
-    return FT_NULLPTR;
-  }
   const const_SockToServIter iter = _socketToServers.find(socket);
   if (iter == _socketToServers.end()) {
-    return FT_NULLPTR;
+    return FT_NULLPTR; // this should never happen
   }
```

Comment:
If this should never happen I think it's much better to `assert()` it.
```suggestion
assert(iter != _socketToServers.end());
```